src/FreqWindow.cpp

```diff
@@ -608,19 +610,30 @@ void FrequencyPlotDialog::GetAudio()
       }
       const auto nChannels = track->NChannels();
       if (track->GetRate() != mRate) {
-         AudacityMessageBox(
-            XO(
-"To plot the spectrum, all selected tracks must be the same sample rate.") );
+         using namespace BasicUI;
+         ShowMessageBox(
+            XO("To plot the spectrum, all selected tracks must be the same sample rate."),
+            MessageBoxOptions {}.Caption(XO("Error")).IconStyle(Icon::Error));
          mData.reset();
          mDataLen = 0;
-         return;
+         return false;
       }
       Floats buffer1{ mDataLen };
       Floats buffer2{ mDataLen };
       float *const buffers[]{ buffer1.get(), buffer2.get() };
       // Don't allow throw for bad reads
-      track->GetFloats(0, nChannels, buffers, start, mDataLen,
-         false, FillFormat::fillZero, false);
+      if (!track->GetFloats(
+             0, nChannels, buffers, start, mDataLen, false,
+             FillFormat::fillZero, false))
+      {
+         using namespace BasicUI;
+         ShowMessageBox(
+            XO("Could not read for analysis. Selecting time-stretched audio is a known reason."),
```

Comment:
"Audio could not be analyzed. This may be due to a stretched clip. Try resetting any stretched clips, or mixing and rendering the tracks before analyzing." might be better

Reply:
That makes me think that we don't have an straightforward way of unstretching clips ... Ticket ?

Reply:
You mean restring the stretch ratio exactly to 1.